core/chain/src/lib.rs

```diff
@@ -123,4 +171,14 @@ where
         }
         Ok(blocks)
     }
+
+    pub fn get_headers_by_index(&self, start: u64, limit: u64) -> Result<Vec<B::SignedHeader>, String> {
+        let mut headers = vec![];
+        for i in start..=(start + limit) {
```

Comment:
It seems that the function intends to get at most `limit` headers, but it actually gets at most `limit+1` headers

Reply:
it gets from [start; start + limit] blocks, e.g. including the sides.

Reply:
So there should be a comment about that. Otherwise people get confused when trying to use the function.